pkgs/by-name/README.md

```diff
@@ -1,9 +1,9 @@
 # Name-based package directories
 
 The structure of this directory maps almost directly to top-level package attributes.
-This is the recommended way to add new top-level packages to Nixpkgs [when possible](#limitations).
+This is how new top-level packages have to be added to Nixpkgs [when possible](#limitations).
 
-Packages found in the named-based structure do not need to be explicitly added to the
+Packages found in the name-based structure must not be added to the
 `top-level/all-packages.nix` file unless they require overriding the default value
 of an implicit attribute (see below).
```

Comment:
I think it is still better to lead with the positive part. Something like

> Packages found in the name-based structure are automatically included, without needing to be added in a file like `all-packages.nix`. They must only occur in file if they require overriding the default value of an implicit attribute (see below).

Reply:
I changed the wording a little bit, but this is essentially applied now :)